klippa/src/lib.rs

```diff
@@ -600,6 +800,50 @@ pub trait Subset {
     ) -> Result<(), SubsetError>;
 }
 
+// A helper trait providing a 'subset' method for various subtables that have no associated tag
+pub(crate) trait SubsetTable {
+    /// Subset this table and write a subset version of this table into serializer
+    fn subset(&self, plan: &Plan, s: &mut Serializer) -> Result<(), SerializeErrorFlags>;
+}
+
+// A trait for subtables that require additional data in order to be subsetted
+pub(crate) trait SubsetArgs {
+    type ArgsForSubset: Default;
+}
+
+pub(crate) trait SubsetTableWithArgs: SubsetArgs {
+    fn subset_with_args(
+        &self,
+        plan: &Plan,
+        s: &mut Serializer,
+        args: &Self::ArgsForSubset,
+    ) -> Result<(), SerializeErrorFlags>;
+}
+
+pub(crate) trait SubsetTwoArgs {
```

Comment:
is this necessary? would it be possible to just have the args be a 2-tuple in some cases? (this is what we do in codegen for `FontReadWithArgs`)